extensions/amp-subscriptions/0.1/deferred-account-flow.js

```diff
@@ -0,0 +1,166 @@
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {Services} from '../../../src/services';
+import {UrlBuilder} from '../../amp-subscriptions/0.1/url-builder';
+import {stringifyForPingback} from '../../../src/utils/xhr-utils';
+
+const LOCAL_STORAGE_KEYS = {
+  HAS_PUBLISHER_ACCOUNT: 'account-exists-on-publisher-side',
+  HAS_REJECTED_ACCOUNT_CREATION: 'user-rejected-account-creation-request',
+};
+/**
+ * AMP-side implementation of deferred account creation, a flow responsible
+ * for reconciling account existance discrepancies across platforms.
+ */
+export class DeferredAccountFlow {
+  /**
+   * @param {!../../../src/service/ampdoc-impl.AmpDoc} ampdoc
+   * @param {string} hasAssociatedAccountUrl
+   * @param {string} accountCreationRedirectUrl
+   * @param {../subscription-platform.SubscriptionPlatform} platform
+   */
+  constructor(
+    ampdoc,
+    hasAssociatedAccountUrl,
+    accountCreationRedirectUrl,
+    platform
+  ) {
+    /** @const @private {!../../../src/service/xhr-impl.Xhr} */
+    this.xhr_ = Services.xhrFor(ampdoc.win);
+    /**
+     * @const @private {function(string):void}
+     * @param {string} url
+     */
+    this.navigateTo_ = (url) => {
+      Services.navigationForDoc(ampdoc).navigateTo(ampdoc.win, url);
+    };
+
+    /** @const @private {!../../../src/service/storage-impl.Storage} */
+    this.storage_ = Services.storageForDoc(ampdoc);
+
+    /** @private @const {?string} */
+    this.hasAssociatedAccountUrl_ = hasAssociatedAccountUrl || null;
+
+    /** @private @const {?string} */
+    this.accountCreationRedirectUrl_ = accountCreationRedirectUrl || null;
+    /**
+     * @private @const {!UrlBuilder}
+     */
+    this.urlBuilder_ = new UrlBuilder(ampdoc, platform.getServiceId());
+
+    /**
+     * @private @const {../subscription-platform.SubscriptionPlatform}
+     */
+    this.platform_ = platform;
+  }
+
+  /**
+   * @return {bool} whether deferred account creation is enabled.
+   */
+  isDeferredAccountCreationEnabled() {
+    return this.hasAssociatedAccountUrl_ && this.accountCreationRedirectUrl_;
+  }
+
+  /**
+   * @param {?./entitlement.Entitlement} selectedEntitlement
+   * @return {!Promise<{found: boolean}>}
+   * @private
+   */
+  hasAssociatedUserAccount_(selectedEntitlement) {
+    let storage;
+    return this.storage_
+      .then((s) => {
+        storage = s;
+        return s.get(LOCAL_STORAGE_KEYS.HAS_PUBLISHER_ACCOUNT);
+      })
+      .then((publisherAccountLocalValue) => {
+        if (publisherAccountLocalValue !== undefined) {
```

Comment:
Should we store and match the swg order id so that a new order cases the account check to run again?